core/src/main/java/org/jclouds/io/internal/BasePayloadSlicer.java

```diff
@@ -50,14 +50,14 @@
 @Singleton
 public class BasePayloadSlicer implements PayloadSlicer {
 
-   public static class PayloadIterator implements Iterable<Payload>, Iterator<Payload> {
+   private static class InputStreamPayloadIterator implements Iterable<Payload>, Iterator<Payload> {
```

Comment:
I changed the name to `InputStreamPayloadIterator` to make this more clear in comparison to `ByteSourcePayloadIterator`.